app/Models/Beatmap.php

```diff
@@ -274,6 +284,34 @@ public function getAttribute($key)
         };
     }
 
+    /**
+     * @return Collection<User>
+     */
+    public function getOwners(): Collection
+    {
+        $owners = $this->beatmapOwners->loadMissing('user')->map(
+            fn ($beatmapOwner) => $beatmapOwner->user ?? new DeletedUser(['user_id' => $beatmapOwner->user_id])
+        );
+
+        // TODO: remove when everything writes to beatmap_owners.
+        if (!$owners->contains(fn ($beatmapOwner) => $beatmapOwner->user_id === $this->user_id) && $this->user !== null) {
+            $owners->prepend($this->user);
+        }
+
+        return $owners;
+    }
+
+    public function isOwner(User $user): bool
```

Comment:
can use similar update for js side (being able to post note and showing "guest" badge on posts)